### PR TITLE
avoid hash for pycbc live

### PR DESCRIPTION
**Pull request by ahnitz**
This may address #3263 for the case of pycbc live. I'm still worried about offline though, so I'll think about that too.

### CODE REVIEW

**Comment by ahnitz** [comment]
Note: The reverse lookup needs to be possible as the template bank is *sliced*. This means the index for __getitem__ won't match the template_id from the original bank. However, we want htilde.id to be the *original* template id, even if the bank was sliced.

**Comment by titodalcanton** [comment]
This seems a very reasonable solution. I will try it out and report.

File: pycbc/waveform/bank.py [hunk 1]
```diff
@@ -490,10 +490,10 @@ def __init__(self, filename, sample_rate, minimum_buffer,
         super(LiveFilterBank, self).__init__(filename, approximant=approximant,
                 parameters=parameters, **kwds)
         self.ensure_standard_filter_columns(low_frequency_cutoff=low_frequency_cutoff)
-        self.hash_lookup = {}
+        self.param_lookup = {}
         for i, p in enumerate(self.table):
-            hash_value =  hash((p.mass1, p.mass2, p.spin1z, p.spin2z))
-            self.hash_lookup[hash_value] = i
+            key =  (p.mass1, p.mass2, p.spin1z, p.spin2z)
+            self.param_lookup[key] = i
```
Comment (titodalcanton) [inline pycbc/waveform/bank.py:496]: Just for good practice against crazy banks, could you add this:
```
assert(key not in self.param_lookup)
```

Comment (ahnitz) [reply pycbc/waveform/bank.py:496]: good idea

**Comment by titodalcanton** [comment]
After this change, I can no longer reproduce issue #3263 with the test script. Please address my comment above and this will be ready for approval.

**Comment by ahnitz** [comment]
@titodalcanton Ok, done

**Pull request merged**